components/post_view/new_messages_below/index.js

```diff
@@ -41,7 +41,7 @@ function makeMapStateToProps() {
         let postIds = getPostIdsInChannel(state, ownProps.channelId);
 
         if (postIds) {
-            postIds = preparePostIdsForPostList(state, {postIds, lastViewedAt: ownProps.lastViewedBottom});
+            postIds = preparePostIdsForPostList(state, {postIds, lastViewedAt: ownProps.lastViewedBottom, channelId: ownProps.channelId});
```

Comment:
Why are we adding channelId here?

Reply:
I added a new selector on redux: https://github.com/Willyfrog/mattermost-redux/blob/2b9fae4819b1349a16ab2c68a9a9cffa44300849/src/utils/post_list.js#L54
this allows the function to know if the user has manually set a post as unread, but we store the mark on the channel.

Reply:
in case you need it, here is the corresponding PR for redux: https://github.com/mattermost/mattermost-redux/pull/903